### PR TITLE
Add VSOCK support

### PR DESCRIPTION
**Pull request by jalil-salame**
I created this patch for my thesis, where I needed to benchmark web servers using VSOCK connections.

I thought someone else might find it useful so I made this PR, but I would understand if you don't want to have the extra optional dependency and dead code.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -169,6 +169,12 @@ Note: If qps is specified, burst will be ignored",
         long = "unix-socket"
     )]
     unix_socket: Option<std::path::PathBuf>,
+    #[cfg(feature = "vsock")]
+    #[clap(
+        help = "Connect to a VSOCK socket instead of the domain in the URL. Only for non-HTTPS URLs.",
```
Comment (hatoo) [inline src/main.rs:174]: Please add a syntax example here. `cid:port`

**Review by hatoo — CHANGES_REQUESTED** [review]

**Comment by hatoo** [comment]
Looks good 👍 
Could you update `CHANGELOG.md` too?

**Comment by jalil-salame** [comment]
Thanks for the quick review c:

**Review by hatoo — APPROVED** [review]

**Comment by hatoo** [comment]
Thank you!

**Pull request merged**